test_conformance/compiler/test_compiler_defines_for_extensions.cpp

```diff
@@ -63,6 +63,7 @@ const char *known_extensions[] = {
     "cl_khr_dx9_media_sharing",
     "cl_khr_egl_event",
     "cl_khr_egl_image",
+    "cl_khr_expect_assume",
```

Comment:
As this isn't an API-only extension, I believe this should go right before `cl_khr_subgroup_rotate`. See also the comments on lines 23 and 56.

Please also update `first_API_extension` accordingly.